### ISSUE TITLE
Make --help work like "help <command>"

### ISSUE THREAD

**Issue by jonathandturner** [nushell/nushell#930]
**Is your feature request related to a problem? Please describe.**
People's natural inclination coming from other shells will likely be to type --help on commands that they know about to learn more about them. We should allow them to use this muscle memory.

**Describe the solution you'd like**
We detect the --help flag, and if found, pass it to the `help` command. If we want to future proof, we may want to retain the rest of the command they typed, but for now just calling `help` on the command they invoked would work best.

This would only work on internal commands, so we'd need to know it was internal before we did this step.

**Describe alternatives you've considered**
We could continue to require people used `help <command>`. This is simple, but not as discoverable and also doesn't allow them to use what they're familiar with.

**Comment by yoshuawuyts** [comment]
If this is useful at all, my first attempt at using nu was something like:

```sh
$ nushell
<command not found>
$ nu
<enter nushell>
$ ls
<directory output>
$ ls -la
<error>
$ ls --help
<invalid flag>
```

This was a somewhat frustrating experience because I now had to search for "how to print help in nushell", and embark on a quest to eventually discover the `help` command. Had nu provided output for `ls --help` by default, it would've saved approx. 10mins from my time onboarding.

**Comment by jonathandturner** [comment]
Fixed with #1226
